src/algorithm/MEMCPY-OMPTarget.cpp

```diff
@@ -30,10 +30,12 @@ namespace algorithm
   int hid = omp_get_initial_device(); \
   int did = omp_get_default_device(); \
 \
-  allocAndInitOpenMPDeviceData(x, m_x, iend, did, hid);
+  allocAndInitOpenMPDeviceData(x, m_x, iend, did, hid); \
+  allocAndInitOpenMPDeviceData(y, m_y, iend, did, hid);
 
 #define MEMCPY_DATA_TEARDOWN_OMP_TARGET \
   deallocOpenMPDeviceData(x, did); \
+  deallocOpenMPDeviceData(y, did);
```

Comment:
Looking at this makes me think we probably need to copy y back to m_y.

Reply:
```suggestion
  getOpenMPDeviceData(m_y, y, iend, hid, did); \
  deallocOpenMPDeviceData(x, did); \
  deallocOpenMPDeviceData(y, did);
```

Reply:
We also don't need the "\" line continuation characters here.

Reply:
I'm not sure what you mean.

Reply:
the backslash. sorry it didn't appear in my comment.

Reply:
nevermind. it's part of the macro. so it's needed. Please disregard my comments.